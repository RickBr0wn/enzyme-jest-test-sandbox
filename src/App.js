import React, { Component } from 'react'
import logo from './logo.svg'
import './App.css'

const Title = ({ title }) => <div>{title}</div>

class App extends Component {
  render() {
    return (
      <div className='App'>
        <header className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className='App-link'
            href='https://reactjs.org'
            target='_blank'
            rel='noopener noreferrer'>
            <h1>Welcome To REACT</h1>
            <Title title={'This is the title prop'} />
          </a>
          <ul className='test-class'>
            <li>1</li>
            <li>1</li>
            <li>1</li>
          </ul>
        </header>
      </div>
    )
  }
}

export default App
