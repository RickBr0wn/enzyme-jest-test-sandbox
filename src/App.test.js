import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('<App/>', () => {
  it('should contain only 1 <p> element', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('p').length).toBe(1)
  })
  it('should contain an element with the className of "App-link"', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('.App-link').exists()).toBe(true)
  })
  it('should contain a <ul> element containing 3 <li> tags', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('ul').children().length).toBe(3)
  })
  it('should contain a <ul> element with the className of "test-class"', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('ul').hasClass('test-class')).toBe(true)
  })
  it('should contain an <h1> element, with the text "Welcome To REACT"', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('h1').text()).toBe('Welcome To REACT')
  })
  it('should contain an <Title/> Component, with the text "This is the title prop"', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('Title').text()).toBe('<Title />')
  })
  it('matches the snapshot', () => {
    const tree = shallow(<App />)
    expect(tree).toMatchSnapshot()
  })
})
