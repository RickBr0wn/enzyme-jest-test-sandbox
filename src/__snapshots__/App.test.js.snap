// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <header
        className="App-header"
      >
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
        />
        <p>
          Edit 
          <code>
            src/App.js
          </code>
           and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h1>
            Welcome To REACT
          </h1>
          <Title
            title="This is the title prop"
          />
        </a>
        <ul
          className="test-class"
        >
          <li>
            1
          </li>
          <li>
            1
          </li>
          <li>
            1
          </li>
        </ul>
      </header>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />,
          <p>
            Edit 
            <code>
              src/App.js
            </code>
             and save to reload.
          </p>,
          <a
            className="App-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h1>
              Welcome To REACT
            </h1>
            <Title
              title="This is the title prop"
            />
          </a>,
          <ul
            className="test-class"
          >
            <li>
              1
            </li>
            <li>
              1
            </li>
            <li>
              1
            </li>
          </ul>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "App-logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Edit ",
              <code>
                src/App.js
              </code>,
              " and save to reload.",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Edit ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "src/App.js",
              },
              "ref": null,
              "rendered": "src/App.js",
              "type": "code",
            },
            " and save to reload.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Welcome To REACT
              </h1>,
              <Title
                title="This is the title prop"
              />,
            ],
            "className": "App-link",
            "href": "https://reactjs.org",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome To REACT",
              },
              "ref": null,
              "rendered": "Welcome To REACT",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "This is the title prop",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                1
              </li>,
              <li>
                1
              </li>,
              <li>
                1
              </li>,
            ],
            "className": "test-class",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "header",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <header
          className="App-header"
        >
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />
          <p>
            Edit 
            <code>
              src/App.js
            </code>
             and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            <h1>
              Welcome To REACT
            </h1>
            <Title
              title="This is the title prop"
            />
          </a>
          <ul
            className="test-class"
          >
            <li>
              1
            </li>
            <li>
              1
            </li>
            <li>
              1
            </li>
          </ul>
        </header>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />,
            <p>
              Edit 
              <code>
                src/App.js
              </code>
               and save to reload.
            </p>,
            <a
              className="App-link"
              href="https://reactjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              <h1>
                Welcome To REACT
              </h1>
              <Title
                title="This is the title prop"
              />
            </a>,
            <ul
              className="test-class"
            >
              <li>
                1
              </li>
              <li>
                1
              </li>
              <li>
                1
              </li>
            </ul>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Edit ",
                <code>
                  src/App.js
                </code>,
                " and save to reload.",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Edit ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "src/App.js",
                },
                "ref": null,
                "rendered": "src/App.js",
                "type": "code",
              },
              " and save to reload.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  Welcome To REACT
                </h1>,
                <Title
                  title="This is the title prop"
                />,
              ],
              "className": "App-link",
              "href": "https://reactjs.org",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome To REACT",
                },
                "ref": null,
                "rendered": "Welcome To REACT",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "title": "This is the title prop",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  1
                </li>,
                <li>
                  1
                </li>,
                <li>
                  1
                </li>,
              ],
              "className": "test-class",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                },
                "ref": null,
                "rendered": "1",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                },
                "ref": null,
                "rendered": "1",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                },
                "ref": null,
                "rendered": "1",
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "header",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
